example/src/example/services/data.cljs

```diff
@@ -1,8 +1,22 @@
-(ns example.services.data)
+(ns example.services.data
+  (:require [venue.core :as venue])
+  (:require-macros [cljs-log.core :as log]))
 
-(defmulti handler (fn [event args] event))
+(defmulti response-handler
+  (fn [result response cursor] result))
 
-(defmethod handler
-  :login
-  [_ args]
-  [:success {:token "foobar"}])
+(defmulti request-handler
+  (fn [event args response-ch] event))
+
+(defn service
```

Comment:
I think you want a deftype here. These anonymous classes are hard to debug.

See https://github.com/cemerick/clojure-type-selection-flowchart